app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:post_id])
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+      redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
+    else
+      flash[:alert] = 'Comment not created!'
+      render :new
+    end
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+end
```

Comment:
- [OPTIONAL] I would recommend you to make your comment_params `private`. You can chek this [link](https://tjoye20.medium.com/ruby-access-control-basics-public-vs-private-vs-protected-methods-7788b26e04a7#:~:text=Private%20methods%20can%20only%20be,receiver%20is%20always%20implicitly%20self.) for more info.

Reply:
Thanks indeed for the feedback and tips.